### ISSUE TITLE
result on sidd

### ISSUE THREAD

**Issue by yarqian** [IDKiro/GMSNet#1]
Many thans for your work. And after restraining the model, I can get similiar result on the dnd benchmark as reported on your paper. But when I trained the model only on the medium version of SIDD, the psnr is  39.10db while the psnr on your paper is 39.51. It seems that there are some problems in my settings. So I am sorry to ask would you please give some advise about that?
Many thanks.

**Comment by IDKiro** [comment]
1. The results submitted to the SIDD benchmark is in uint8, so use 'np.round()' when you convert float data to uint8 data
2. 8 directions self-ensemble +  larger input patch (refer to utils/dndsubmit/dnd_denoise.py)
3. Using epoch-ensemble and model-ensemble when submitting results to SIDD benchmark.
Our result is obtained by online evaluation: https://www.eecs.yorku.ca/~kamel/sidd/benchmark.php
I don't know if your result is the result of online evaluation or the result on the local validation set.

**Comment by yarqian** [comment]
Many thanks for your reply.
1. The images  have been converted into uint8 data.
2. 8 directions self-ensemble has been used, but the input size is only 128x128, so I did not use the larger input patch
3.  I have not used the epoch-ensemble and model-ensemble, but it seems that such methods cannont improve the result with 0.4db.
Actually the difference bwteen the result of the online evaluation and the local validation set is small. The gap is about 0.03 level. 
I am sorry to ask is it possible that you can share the code for training and testing. My email is qian@vision.is.tohoku.ac.jp. 
Many thanks.

**Comment by yarqian** [comment]
Many thanks for your reply.

The images have been converted into uint8 data.
8 directions self-ensemble has been used, but the input size is only 128x128, so I did not use the larger input patch
I have not used the epoch-ensemble and model-ensemble, but it seems that such methods cannont improve the result with 0.4db.
Actually the difference bwteen the result of the online evaluation and the local validation set is small. The gap is about 0.03 level.
I am sorry to ask is it possible that you can share the code for training and testing for the SIDD benchmark. My email is qian@vision.is.tohoku.ac.jp.
Many thanks.

**Comment by IDKiro** [comment]
SIDD benchmark provides original test images, so you can crop larger patches by yourself.
Make sure that you use 'np.round()' when converting the data to uint8.

Unfortunately I cannot provide SIDD training and submit code since the SDD of my PC went down last month. In order to speed up the image loading, all the data of GMSNet was stored on the SSD at that time. So if you use the pre-trained model I provided to submit to DND, you will find that your scores are inconsistent with those in the paper, because the code of this repo are rewritten after the data is lost.

**Comment by IDKiro** [comment]
If you are interested in the SIDD benchmark, you can follow the work of my friend
https://github.com/happycaoyue/
PseudoISP(PT-MWRN), HT-MWResnet and mwresnet are all his work.

**Comment by yarqian** [comment]
Got it. Many thanks for your help and I will close this issue.
